puppet/spec/unit/indirector/catalog/puppetdb_spec.rb

```diff
@@ -709,14 +709,41 @@ def test_file_require(resource_title, require_title)
           'transaction_uuid', 'environment', 'producer_timestamp', "code_id",
           "job_id", "catalog_uuid", "producer"]
       end
+
+      context 'when dealing with TagSets' do
+        let(:resource_hashes) do
+          {
+            'resources' => [
+              {'type' => 'Stage',
+              'title' => :main,
+              'tags' => Puppet::Util::TagSet.new(['stage']),
+              'exported' => false,
+              'parameters' => {:name => 'main'}},
+             {'type' => 'Class',
+              'title' => 'Settings',
+              'tags' => Puppet::Util::TagSet.new(['class', 'settings']),
+              'exported' => false},
+             {'type' => 'Class',
+              'title' => :main,
+              'tags' => Puppet::Util::TagSet.new(['class']),
+              'exported' => false,
+              'parameters' => {:name => 'main'}}
+            ]
+          }
+        end
+
+        it "should convert TagSet for resource['tags']" do
+          subject.hashify_tags(resource_hashes)['resources'].each { |resource| resource['tags'].should be_an Array }
```

Comment:
Minor concern - I'd prefer to see this wrapped a bit since, if nothing else, it "goes off the side" of github's diff view; don't think we have a pdb standard, but maybe Russ and I can compromise on "normally under 100").